### PR TITLE
Feature/thrust calculator quadro copter

### LINKED ISSUE #41: Thrust Calculator for QuadroCopter
**Describe the solution you'd like**
This should derive from `ThrustCalculator` and should calculate the thrust for four rotors depending on `DroneMetrics` and a current and a future `DroneState`.

### PR DESCRIPTION
**Pull request by Segelzwerg**
closes #41

### CODE REVIEW

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/calculation_layer/ThrustCalculator.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5305011&bid=17363445)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/calculation_layer/ThrustCalculator.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5305011&bid=17363445)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/calculation_layer/thrust_calculator.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5305011&bid=17363445)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/calculation_layer/thrust_calculator.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5305011&bid=17363445)

File: src/drone_controller/calculation_layer/thrust_calculator.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        self.radius = radius
+
+    def calc(self, current_state: DroneState, future_state: DroneState) -> list:
+        """
+        The actual calculation.
+        :param current_state: the state of drone now
+        :param future_state: the desired state of the drone
+        :return: four relative thrusts values (for DC rotors)
+        """
+        jerk = self.subtract_lists(future_state.state_dict['Acceleration'],
```
Comment (TreeKid) [inline src/drone_controller/calculation_layer/thrust_calculator.py:44]: Add comment why acc -acc  can be  jerk.

File: src/drone_controller/calculation_layer/thrust_calculator.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class ThrustCalculatorQuadroCopter(ThrustCalculator):
+    """
+    Calculates thrusts for a quadro copter.
+    """
+
+    def __init__(self, mass: float, rotor_thrust: float, radius: float):
+        self.mass = mass
+        self.rotor_thrust = rotor_thrust
```
Comment (TreeKid) [inline src/drone_controller/calculation_layer/thrust_calculator.py:34]: rename to rotor max thrust

File: src/drone_controller/calculation_layer/thrust_calculator.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        """
+        raise NotImplementedError
+
+
+class ThrustCalculatorQuadroCopter(ThrustCalculator):
+    """
+    Calculates thrusts for a quadro copter.
+    """
+
+    def __init__(self, mass: float, rotor_thrust: float, radius: float):
```
Comment (TreeKid) [inline src/drone_controller/calculation_layer/thrust_calculator.py:32]: Füge Dämpfungs Parameter hinzu um die Trägheit der Drohne auszugleichen.

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/calculation_layer/thrust_calculator.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5305011&bid=17363445)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/calculation_layer/thrust_calculator.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5305011&bid=17363445)

**Pull request merged**